### DISCUSSION TITLE
Vitest working example & few questions

### DISCUSSION THREAD

**Discussion by r9mp** [nuxt/nuxt#16678] [General]
Hi guys,

I had lots of difficulties to implement vitest with vue 3 and Nuxt 3. So here is a working example: https://github.com/r9mp/vitest-nuxt

If you have improvment ideas or if I did something to complex, don't hesitate, I will keep this repo / message updated!

And if this example seems ok for you, I'll add it to Vitest doc examples.

**Comment by kingtimm** [comment]
Thanks for taking the time to do this. Leaving comments in the vite config about what the imports get us was also super helpful 😄

**Comment by adrianrivers** [comment]
Hi, I have a similar setup to the one in the repository, I haven't managed to figure out how to render `NuxtLink` components in the tests, currently it gets output as a custom element e.g. `<nuxt-link to="/foo">`. Has anyone figured out how to correctly output native Nuxt components in tests using a similar setup as the one in the repository?
